src/readers/streams_reader.c

```diff
@@ -84,7 +84,7 @@ typedef struct SingleStreamReaderCtx{
     long long pendingMessages;
     long long nextBatch;
     bool isRunning;
-
+    bool freeWhenDone;
```

Comment:
Nit: Maybe isFreeWhenDone to match isRunning?

Reply:
Fixed